File: python/login_signup.py
def check_login(user,pwd):
	if user in data and data[user]["password"] == pwd:
		return True
	else:
		return False
	
def check_validation(user,pwd,confirm_pwd):
	if pwd == confirm_pwd and user not in data:
		data[user] = {"password":pwd}
		return True
	else:
		return False


data = {"user": {"password": "pass"}}

logged = False
username = ""

default_text = """[1] Login
[2] Signup
"""

while True:
	while not logged:
		print(default_text)
		option = input("Choose: ")
		
		if option == "1":
			user = input("Username: ")
			pwd = input("Password: ")
			check = check_login(user,pwd)
			
			if check:
				username, logged = user, True
			else:
				print("Invalid username or password")
		elif option == "2":
			user = input("Username: ")
			pwd = input("Password: ")
			confirm_pwd = input("Confirm password: ")
			valid = check_validation(user,pwd,confirm_pwd)
			if valid:
				username,logged = user, True
				print("Account created succesfully")
			elif user in data:
				print("Username already exist!")
			else:
				print("Password doesnt matched")
			
		else: print("Invalid option")	
	while logged:
		print(f"Hello, {username}!")
		print("[1] Logout\n")
		option = input("Choose: ")
		
		if option == "1": username,logged = "",False
		else: print("Invalid input!")
		
		

File: java/login_signup.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class UserAuthentication {

    static Map<String, Map<String, String>> data = new HashMap<>();
    static boolean logged = false;
    static String username = "";

    public static boolean checkLogin(String user, String pwd) {
        return data.containsKey(user) && data.get(user).get("password").equals(pwd);
    }

    public static boolean checkValidation(String user, String pwd, String confirmPwd) {
        if (pwd.equals(confirmPwd) && !data.containsKey(user)) {
            data.put(user, new HashMap<>());
            data.get(user).put("password", pwd);
            return true;
        } else {
            return false;
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String defaultText = "[1] Login\n[2] Signup\n";

        while (true) {
            while (!logged) {
                System.out.println(defaultText);
                System.out.print("Choose: ");
                String option = scanner.nextLine();

                if (option.equals("1")) {
                    System.out.print("Username: ");
                    String user = scanner.nextLine();
                    System.out.print("Password: ");
                    String pwd = scanner.nextLine();
                    boolean check = checkLogin(user, pwd);

                    if (check) {
                        username = user;
                        logged = true;
                    } else {
                        System.out.println("Invalid username or password");
                    }
                } else if (option.equals("2")) {
                    System.out.print("Username: ");
                    String user = scanner.nextLine();
                    System.out.print("Password: ");
                    String pwd = scanner.nextLine();
                    System.out.print("Confirm password: ");
                    String confirmPwd = scanner.nextLine();
                    boolean valid = checkValidation(user, pwd, confirmPwd);

                    if (valid) {
                        username = user;
                        logged = true;
                        System.out.println("Account created successfully");
                    } else if (data.containsKey(user)) {
                        System.out.println("Username already exists!");
                    } else {
                        System.out.println("Passwords don't match");
                    }
                } else {
                    System.out.println("Invalid option");
                }
            }

            while (logged) {
                System.out.println("Hello, " + username + "!");
                System.out.println("[1] Logout\n");
                System.out.print("Choose: ");
                String option = scanner.nextLine();

                if (option.equals("1")) {
                    username = "";
                    logged = false;
                } else {
                    System.out.println("Invalid input!");
                }
            }
        }
    }
                }
